samples/99_zil_interpreter/app/builtins.rb

```diff
@@ -16,6 +16,106 @@ def define_for_evaled_arguments(&implementation)
   context.locals[var_atom]
 }
 
+# <+ ...>
 ZIL_BUILTINS[:+] = define_for_evaled_arguments { |arguments|
   arguments.inject(0, :+)
 }
+
+# <- ...>
+ZIL_BUILTINS[:-] = lambda { |arguments, context|
```

Comment:
You can the `define_for_evaled_arguments` helper in place of lambda..... for `SUBR`s - i.e. functions that have all their arguments evaluated.... it takes care of that boilerplate stuff for you

Reply:
These should be better now.